pint_server/app.py

```diff
@@ -38,6 +41,32 @@
                                 VersionsModel, MicrosoftRegionMapModel)
 
 
+# hashable helper class to create named tuples from the details in
+# an image entry that are relevant for deletion date tracking.
+DeletionDetails = namedtuple("DeletionDetails",
+                             " ".join(["state",
+                                       "deprecatedon",
+                                       "deletedon"]))
+
+
+# helper regexp matcher for dates specified in the %Y%m%d (4 digits
+# for year, 2 digits for month, 2 digits for day) format
+date_matcher = re.compile(
```

Comment:
Instead of regex, why not just let `datetime.strptime()` handle the format checking?

Reply:
Because when I called it with, for example `111111` it comes back with year=1111, month=1, day=1... So the regex helps with ensuring we only accept a valid 8 digit date spec.

```
>>> import datetime
>>> DATE_FORMAT = '%Y%m%d'
>>> datetime.datetime.strptime('11111111', DATE_FORMAT)
datetime.datetime(1111, 11, 11, 0, 0)
>>> datetime.datetime.strptime('111111', DATE_FORMAT)
datetime.datetime(1111, 1, 1, 0, 0)
>>> datetime.datetime.strptime('1111111', DATE_FORMAT)
datetime.datetime(1111, 11, 1, 0, 0)
>>> datetime.datetime.strptime('11111111', DATE_FORMAT)
datetime.datetime(1111, 11, 11, 0, 0)
>>> datetime.datetime.strptime('11110101', DATE_FORMAT)
datetime.datetime(1111, 1, 1, 0, 0)
>>> datetime.datetime.strptime('1111101', DATE_FORMAT)
datetime.datetime(1111, 10, 1, 0, 0)
>>> datetime.datetime.strptime('1111011', DATE_FORMAT)
datetime.datetime(1111, 1, 1, 0, 0)
```